saml/src/main/java/com/linecorp/armeria/server/saml/KeyStoreCredentialResolverBuilder.java

```diff
@@ -56,15 +67,32 @@ public KeyStoreCredentialResolverBuilder addKeyPassword(String name, String pass
         return this;
     }
 
+    /**
+     * Adds all key names and their passwords which are specified by the {@code keyPasswords}.
+     */
+    public KeyStoreCredentialResolverBuilder addKeyPasswords(Map<String, String> keyPasswords) {
+        requireNonNull(keyPasswords, "keyPasswords");
+        keyPasswords.forEach(this::addKeyPassword);
+        return this;
+    }
+
     /**
      * Creates a new {@link KeyStoreCredentialResolver}.
      */
     public CredentialResolver build() throws IOException, GeneralSecurityException {
-        final KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
-        try (InputStream is = KeyStoreCredentialResolverBuilder.class.getClassLoader()
-                                                                     .getResourceAsStream(path)) {
+        final KeyStore ks = KeyStore.getInstance(type);
+        try (InputStream is = open(path)) {
             ks.load(is, keystorePassword.toCharArray());
         }
         return new KeyStoreCredentialResolver(ks, keyPasswords);
     }
+
+    private static InputStream open(String path) throws IOException {
+        final File file = new File(path);
+        if (file.isFile() && file.exists()) {
```

Comment:
Instead of auto-detection, how about providing two constructors? One that accepts `File` or `Path` and the other that accepts `String path` and `ClassLoader`